src/core/namespace.c

```diff
@@ -775,12 +775,11 @@ static void drop_nop(MountEntry *m, size_t *n) {
                         bool found = false;
 
                         /* Now let's find the first parent of the entry we are looking at. */
-                        for (p = t-1; p >= m; p--) {
+                        for (p = PTR_SUB1(t, m); p; p = PTR_SUB1(p, m))
```

Comment:
In general, I'd like to split the iterator and `found` entry, and make the iterator defined in `for`. How about the following?
```
MountEntry *found = NULL;

for (MountEntry *p = PTR_SUB1(t, m); p; p = PTR_SUB1(p, m))
        if (path_startswith(...)) {
                found = p;
                break;
        }
```